libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/usecase/route/FetchRouteUseCase.kt

```diff
@@ -0,0 +1,31 @@
+package com.mapbox.navigation.dropin.usecase.route
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.extensions.coroutines.requestRoutes
+import com.mapbox.navigation.dropin.usecase.UseCase
+import com.mapbox.navigation.dropin.usecase.location.GetCurrentLocationUseCase
+import com.mapbox.navigation.utils.internal.toPoint
+import kotlinx.coroutines.CoroutineDispatcher
+
+class FetchRouteUseCase(
+    private val navigation: MapboxNavigation,
```

Comment:
Why not make these `MapboxNavigationObserver`?

The `MapboxNavigation` instance can change at any time. So these use cases need to be re-instantiated?

Reply:
These classes encapsulate business logic that drive the system. They should be lightweight, stateless and only execute their logic when invoked.

`MapboxNavigation` is this class direct dependency. When it changes, this class must be reinstantiated. If we find that instantiation of this class affects system performance we will address it by injecting `Provider<MapboxNavigation>` class that will return currently attached `MapboxNavigation` during execution.